classes/auth.php

```diff
@@ -22,492 +24,479 @@ class AuthException extends \FuelException {}
+                            return true;
+                        }
+                    }
+                }
+            } else {
+                foreach (static::$_instances as $i) {
+                    if ($i->$method($condition, null, $entity)) {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        } else {
+            if ($entity === null) {
+                foreach (static::$_verified as $v) {
+                    if (static::$type($driver)->$method($condition)) {
+                        return true;
+                    }
+                }
+            } elseif (static::$type($driver)->$method($condition, $entity)) {
+                return true;
+            }
+
+            return false;
+        }
+    }
+
+    public static function has_role($role_id = null, $user = null)
+    {
+        if (! is_array($role_id)) {
+            $role_id = array($role_id);
+        }
+
+        if ($user === null && static::instance()) {
+            $user = static::get_user();
+        }
+
+        if ($user instanceof Model\Auth_User && ($user->roles || $user->group)) {
+            $roles = $user->roles;
+            if ($user->group && $user->group->roles) {
+                $roles = array_merge($roles, $user->group->roles);
+            }
+
+            foreach ($roles as $role) {
+                foreach ($role_id as $test_role) {
+                    $tocheck = $role->name;
+
+                    // asterisk means "any of"
+                    if (substr($test_role, -2) === '.*') {
+                        // remove down to full stop on test_role
+                        $test_role = substr($test_role, 0, -1);
+
+                        // don't worry about the full stop, just go for a match against test role
+                        // anything more would be overkill
+                        $tocheck = substr($tocheck, 0, strlen($test_role));
+                    }
+
+                    if ($tocheck == $test_role) {
+                        return true;
+                    }
+                }
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Check if an INDEX exists on a table.
+     * This is a hack becaues DBUtil doesn't have this method and we
+     * shouldn't need to rely on a particular version of fuel-core
+     * @param  string $table_name      The table name to check on
+     * @param  string $index_name The index name to check for
+     * @return boolean			  Returns true if exists, false if doesn't
+     */
+    public static function check_index_exists($table_name, $index_name)
+    {
+        $dsn = \Config::get('db.'.\Config::get('db.active').'.connection.dsn');
+        foreach (explode(';', $dsn) as $dsn_part) {
+            $exploded = explode('=', $dsn_part);
+            if ($exploded[0] == 'dbname') {
+                $database_name = $exploded[1];
+                break;
+            }
+        }
+
```

Comment:
should probably check `$database_name` got set even though its highly likely that it did get set?

Reply:
Now throwing an exception